src/main/java/org/openrewrite/java/testing/junit4toassertj/ChangeAssertTrue.java

```diff
@@ -0,0 +1,124 @@
+package org.openrewrite.java.testing.junit4toassertj;
+
+import org.openrewrite.internal.lang.Nullable;
+import org.openrewrite.java.*;
+import org.openrewrite.java.tree.Expression;
+import org.openrewrite.java.tree.J;
+import org.openrewrite.java.tree.JavaType;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import static org.openrewrite.Formatting.EMPTY;
+import static org.openrewrite.Formatting.format;
+import static org.openrewrite.Tree.randomId;
+
+/**
+ * This is a refactoring visitor that will convert JUnit-style assertTrue() to assertJ asserts:
+ *
+ * assertTrue("This should be true", EXPRESSION) -> assert(EXPRESSION).with("This should be true").isTrue()
+ * assertTrue(EXPRESSION) -> assert(EXPRESSION).isTrue()
+ *
+ */
+public class ChangeAssertTrue extends JavaIsoRefactorVisitor {
+
+    private static final MethodMatcher assertTrueMatcher = new MethodMatcher("org.junit.Assert assertTrue(..)");
+
+    public ChangeAssertTrue() {
+        setCursoringOn();
+    }
+
+    @Override
+    public J.CompilationUnit visitCompilationUnit(J.CompilationUnit cu) {
+
+        maybeRemoveImport("org.junit.Assert");
+        return super.visitCompilationUnit(cu);
+    }
+
+    @Override
+    public J.MethodInvocation visitMethodInvocation(J.MethodInvocation method) {
+        J.MethodInvocation original = super.visitMethodInvocation(method);
+        if (!assertTrueMatcher.matches(method)) {
+            return original;
+        }
+
+        //The expression that evaluates to true/false is either the first argument (if using the one-argument form)
+        //or the second argument (if using the two-argument form)
+
+        List<Expression> originalArgs = original.getArgs().getArgs();
+        Expression assertExpression = originalArgs.size() == 1 ? originalArgs.get(0) : originalArgs.get(1);
+
+        //This create the `assertThat(<EXPRESSION>)` method invocation.
+        J.MethodInvocation assertSelect = new J.MethodInvocation(
+                randomId(),
+                null,
+                null,
+                J.Ident.build(randomId(), "assertThat", JavaType.Primitive.Void, EMPTY),
+                new J.MethodInvocation.Arguments(
+                        randomId(),
+                        Collections.singletonList(assertExpression.withPrefix("")),
+                        EMPTY
+                ),
+                null,
```

Comment:
Leaving the `type` parameter of a `J.MethodInvocation` `null` can have some non-obvious consequences.
If you immediately serialize it back to text everything is OK. But if other visitors stop by they might be confused. The most likely visitor to stop by, be confused, and actually mess things up is `OrderImports` with `removeUnused` set to `true`. That visitor will look at this, see that the type is null, and happily remove the import for `assertThat` because it didn't find any usages. 

Constructing `JavaType.Method` is a bit annoying, but so long as you get the fully qualified name of the class in there it's much less likely that any subsequent other visitor will have troubles here. You may generally find it more convenient to use `treeBuilder` to construct method invocations for this reason.

Reply:
Hah, I ran into exactly this issue with my "maybeAddStaticImport()" method. I did have `setOnlyIfReferenced` set to true but it was not adding my import. This makes total sense now that you are describing it. I also wasn't sure how to correctly setup the type but I like the idea of exploring the tree builder, I will give that a spin tonight.